iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImpl.java

```diff
@@ -115,11 +116,12 @@ public StringLiteral version() {
         return new StringLiteralImpl(token);
       }
     }
-    throw new UnsupportedOperationException("The typeAndVersion field should contains `type@version` value, but it's " + text);
+    return typeAndVersion;
```

Comment:
I had to remove this exception throw because it currently crashes the parser and prevents it from parsing the file + all projects.
It is valid in bicep to not have a version in a resource declaration apparently, tickets created to fix this: [SONARIAC-1018](https://sonarsource.atlassian.net/browse/SONARIAC-1018) and [SONARIAC-1019](https://sonarsource.atlassian.net/browse/SONARIAC-1019).